### ISSUE TITLE
App bundle notarization for Mac OS

### ISSUE THREAD

**Issue by dryarullin** [Nuitka/Nuitka#2232]
I'm using `--standalone` option to speed up application startup time. I also create a bundle using` --macos-create-app-bundle` option.
But I also need to sign and notarize the application. There were problems with this.

**nuitka3 --version**

1.5.7
Commercial: None
Python: 3.11.0 (main, Mar 1 2023, 12:49:28) [Clang 14.0.6 ]
Flavor: Anaconda Python
Executable: /Users/dryarullin/miniconda3/envs/pyqt_new/bin/python
OS: Darwin
Arch: x86_64
Version C compiler: /usr/bin/clang (clang).

**Command (main part)**

1. export version="$(cat $root/_version.py | awk -F "= " '{print $2}' | xargs)"
export APP_NAME="$(cat $root/config.py | awk -F "APP_NAME = " '{print $2}' | xargs)"
export APP_ID_DARWIN="$(cat $root/config.py | awk -F "APP_ID_DARWIN = " '{print $2}' | xargs)"
nuitka3 --standalone \
        --plugin-enable=pyside6 --include-data-dir=./configs=configs \
        --include-plugin-files=./package/mac/imports.py \
        --include-qt-plugins=qml,platforms,imageformats --output-file=${APP_NAME} --output-dir=dist \
        --macos-create-app-bundle --macos-app-mode=gui --macos-app-icon=./images/fluenta-app_icon.icns \
        --macos-signed-app-name=${APP_ID_DARWIN} --disable-console --macos-app-version=${version} \
        --macos-app-protected-resource="NSMicrophoneUsageDescription:This app requires access to microphone" \
           main.py

2. codesign --verify --force --verbose --deep --options runtime --timestamp --entitlements someinfo.plist --sign "$CERTIFICATE_NAME_APPLICATION" $APP_BUNDLE.app

I know that nuitka can sign the bundle, but I need additional entitlements. Thus, the bundle was re-sign here. The signature verification is successful.

The next step is notarization. This is an deprecated type of notarization, but still working.

3. xcrun altool --notarize-app --type osx \
    --primary-bundle-id $APP_BUNDLE_ID \
    --username $APPLE_ID_USER --password "@keychain:my_app_pass" \
    --file $APP_BUNDLE.app --output-format xml

Notarization fails. Error description:
```
{
  "logFormatVersion": 1,
  "jobId": "",
  "status": "Invalid",
  "statusSummary": "Archive contains critical validation errors",
  "statusCode": 4000,
  "archiveFilename": "myapp-macos-installer-x64-0.1.0.0.pkg",
  "uploadDate": "2023-05-16T18:55:56Z",
  "sha256": "",
  "ticketContents": null,
  "issues": [
    {
      "severity": "error",
      "code": null,
      "path": "myapp-macos-installer-x64-0.1.0.0.pkg/myapp.pkg Contents/Payload/Applications/myapp.app/Contents/MacOS/myapp",
      "message": "The signature of the binary is invalid.",
      "docUrl": "https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/resolving_common_notarization_issues#3087735",
      "architecture": "x86_64"
    }
  ]
}
```

I found what is the problem. `myapp.app/Contents/MacOS/myapp` folder must contain only executable files (https://developer.apple.com/forums/thread/680243).

Is there a way to move the contents of `myapp.app/Contents/MacOS/myapp` to another location, leaving only the `myapp `file?

**Comment by kayhayen** [comment]
If you let Nuitka sign, it's supposed to work I think. And it can also add entitlements while signing. Can you retry with that? I believe you are probably not signing yourself properly, and I won't want to investigate if that is the case.

**Comment by kayhayen** [comment]
The onefile on macOS is not supposed to be used with bundle mode, and is not going to preserve entitlements at all. Nuitka can create them, and for a few packages, it's a workaround, but a bad one.

I may have observed, that signing options are available even when not actually signing. That is a bad thing, and a bug. Can you confirm or deny to having used then `--macos-create-app-bundle` option. Without that one, signing it not happening, and it's not signable. I don't think anything but apps can be signed really, and definitely not notarized.

**Comment by dryarullin** [comment]
Without `--macos-create-app-bundle` option, I get an error for both `--onefile` and one `--standalone`. Error description:` FATAL: options-nanny: Error, package 'Foundation' requires '--macos-create-app-bundle' to be used or else it cannot work.`

I found a solution to the problem for myself. I am compiling a bundle (`--standalone --macos-create-app-bundle`) without code sign options. Then I moved the folders with text files, pictures and qml files to another location. For example, ./Contents/MacOS/PySide6 -> ./Contents/lib/PySide6. One of the folders even contained hidden text file. It also caused problems during notarization. Then I created relative symlinks to the moved folders. I signed all the files in the moved folders, except for the picture, text, qml files (only *.so, *.dylib). Then I signed the bundle. After that, the notarization was successful. Signing files that don't contain code is a bad idea.
